### PR TITLE
hud: print docker compose logs to log modal

### PR DESCRIPTION
**Pull request by jazzdan**
Currently looks something like this:

screen shot 2018-12-20 at 5 36 29 pm

### CODE REVIEW

File: internal/engine/docker_compose_log_manager.go [hunk 1]
```diff
@@ -142,7 +142,7 @@ type DockerComposeLogActionWriter struct {
 func (w DockerComposeLogActionWriter) Write(p []byte) (n int, err error) {
 	w.store.Dispatch(DockerComposeLogAction{
 		ManifestName: w.manifestName,
-		Log:          append([]byte{}, p...),
```
Comment (jazzdan) [inline internal/engine/docker_compose_log_manager.go]: This seems unnecessary to me?

File: internal/engine/upper.go [hunk 2]
```diff
@@ -197,7 +197,7 @@ func handleBuildStarted(ctx context.Context, state *store.EngineState, action Bu
 	}
 
 	if dcState := ms.DCResourceState(); !dcState.Empty() {
-		ms.ResourceState = dcState.WithCurrentLog([]byte{}) // TODO(maia): when reset(/not) CrashLog for DC service?
+		ms.ResourceState = dcState.WithCurrentLog(dcState.CurrentLog) // TODO(maia): when reset(/not) CrashLog for DC service?
```
Comment (jazzdan) [inline internal/engine/upper.go]: @maiamcc can you elaborate on this comment?

File: internal/engine/docker_compose_log_manager.go (hunk 1 above)
Comment (nicks) [reply internal/engine/docker_compose_log_manager.go]: from the interface docs:
https://golang.org/pkg/io/#Writer
"Implementations must not retain p"

slices are mutable, so you don't have any real guarantee that this byte slice will be the same later, when the action is processed

Comment (jazzdan) [reply internal/engine/docker_compose_log_manager.go]: aha! Thank you, I learned something new. :)

File: internal/hud/renderer.go [hunk 3]
```diff
@@ -196,14 +196,9 @@ func isCrashing(res view.Resource) bool {
...
 func bestLogs(res view.Resource) string {
-	// TODO(maia): get the BEST logs (currently getting any old logs)
-	if res.IsDC() {
-		lastBuildLog := strings.TrimSpace(string(res.LastBuild().Log))
-		if res.LastBuild().StartTime.Before(res.PodCreationTime) &&
-			len(lastBuildLog) > 0 {
-			return lastBuildLog + "\n\n" + res.PodLog
-		}
-		return res.PodLog
+	// TODO(maia): get the BEST logs (currently getting all logs)
```
Comment (maiamcc) [inline internal/hud/renderer.go]: again, b/c we're not wrestling with pods that come and go, I think the "best" log for a dc service is just... it's log. I'll remove this comment.

**Pull request merged**